cli/src/pcluster/templates/login_nodes_stack.py

```diff
@@ -0,0 +1,140 @@
+                self,
+                f"LoginNodeIamResources{pool.name}",
+                self._config,
+                pool,
+                self._shared_storage_infos,
+                pool.name,
+            )
+        self._login_nodes_instance_profiles = {k: v.instance_profile for k, v in iam_resources.items()}
+        self.managed_login_nodes_instance_roles = {k: v.instance_role for k, v in iam_resources.items()}
+
+    def _add_launch_templates(self):
+        self.login_launch_templates = {}
+        for pool in self._login_nodes.pools:
+            pool_lt_security_groups = get_login_nodes_security_groups_full(
+                self._login_security_group,
+                pool,
+            )
+            self.login_launch_templates[pool.name] = self._add_login_nodes_pool_launch_template(
+                pool,
+                pool_lt_security_groups,
+                self._login_nodes_instance_profiles,
+            )
+
+    def _add_auto_scaling_groups(self):
+        self.login_nodes_auto_security_groups = {}
+        for pool in self._login_nodes.pools:
+            self.login_nodes_auto_scaling_groups[pool.name] = self._add_login_nodes_pool_auto_scaling_group(
+                pool,
+            )
+
+    def _add_login_nodes_pool_launch_template(
+        self,
+        login_nodes_pool: LoginNodesPools,
+        login_nodes_pool_lt_security_groups,
+        login_nodes_instance_profiles,
+    ):
+        login_node_lt_nw_interface = [
+            ec2.CfnLaunchTemplate.NetworkInterfaceProperty(
+                device_index=0,
+                interface_type=None,
+                groups=login_nodes_pool_lt_security_groups,
+                subnet_id=login_nodes_pool.networking.subnet_id
+            )
+        ]
+        return ec2.CfnLaunchTemplate(
+            self,
+            f"LoginNodeLaunchTemplate{login_nodes_pool.name}",
+            launch_template_name=f"{self.stack_name}-{login_nodes_pool.name}",
+            launch_template_data=ec2.CfnLaunchTemplate.LaunchTemplateDataProperty(
+                image_id=login_nodes_pool.image.custom_ami,
+                instance_type=login_nodes_pool.instance_type,
+                key_name=login_nodes_pool.ssh.key_name,
+                iam_instance_profile=ec2.CfnLaunchTemplate.IamInstanceProfileProperty(
+                    name=login_nodes_instance_profiles[login_nodes_pool.name]
+                ),
+                metadata_options=ec2.CfnLaunchTemplate.MetadataOptionsProperty(
+                    http_tokens=get_http_tokens_setting(self._config.imds.imds_support)
+                ),
+                network_interfaces=login_node_lt_nw_interface,
+                # user_data=
+                # block_device_mappings=
+            )
+        )
+
+    def _add_login_nodes_pool_auto_scaling_group(
+        self,
+        login_nodes_pool: LoginNodesPools,
+    ):
+        auto_scaling_group = autoscaling.AutoScalingGroup(
```

Comment:
I recommend using an L1 Construct ([CfnAutoScalingGroup](https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_autoscaling/CfnAutoScalingGroup.html)) since the ASG will use a LaunchTemplate to describe the login node instances.